pkg/mimirtool/commands/analyse_prometheus.go

```diff
@@ -34,173 +38,197 @@ type PrometheusAnalyzeCommand struct {
 	grafanaMetricsFile string
 	rulerMetricsFile   string
 	outputFile         string
+	concurrency        int
 }
 
 func (cmd *PrometheusAnalyzeCommand) run(k *kingpin.ParseContext) error {
+	metricsUsed, err := cmd.parse()
+	if err != nil {
+		return err
+	}
+
+	v1api, err := cmd.init()
+	if err != nil {
+		return err
+	}
+
+	apiWithBackoff := withBackoff(v1api)
+
+	metricNames, err := cmd.queryMetricNames(apiWithBackoff)
+	if err != nil {
+		return err
+	}
+
+	log.Debugln("Starting to analyze metrics in use")
+	inUseMetricsAnalyse, err := cmd.analyze(apiWithBackoff, metricsUsed, func(model.LabelValue) bool { return false })
+	if err != nil {
+		return err
+	}
+	log.Infof("%d active series are being used in dashboards", inUseMetricsAnalyse.cardinality)
+
+	log.Debugln("Starting to analyze additional metrics in use")
+	additionalMetricsAnalyse, err := cmd.analyze(apiWithBackoff, metricNames, func(metrics analyse) func(metric model.LabelValue) bool {
+		return func(metric model.LabelValue) bool {
+			_, ok := metrics.stats[metric]
+			return ok
+		}
+	}(inUseMetricsAnalyse))
```

Comment:
```suggestion
	additionalMetricsAnalyse, err := cmd.analyze(apiWithBackoff, metricNames, func(metric model.LabelValue) bool {
			_, ok := inUseMetricsAnalyse.stats[metric]
			return ok
	})
```

This can be simplified to avoid the second closure, right?